config/dataprovider.go

```diff
@@ -49,3 +51,167 @@ func GetDynamicString(dnVal string, dP DataProvider) (string, error) {
 	}
 	return dnVal, nil
 }
+
+//NewObjectDP constructs a DataProvider
+func NewObjectDP(obj interface{}) (dP DataProvider) {
```

Comment:
Is the ObjectDP needed here, or it can run in utils? If it is mandatory here, please rename the file to objdp.go